sentinel/src/lib/scheduling.js

```diff
@@ -0,0 +1,45 @@
+const later = require('later');
+
+/**
+ * Fetch a schedule based on the configuration, parsing it as a "cron"
+ * or "text" statement see:
+ * http://bunkat.github.io/later/parsers.html
+ *
+ * @param {Object} config
+ * @param {string} config.text_expression a text expression, like 'every 12 hours'
+ * @param {string} config.cron a cron expression, like '* 1 * * *'
+ * @return {Object} the schedule object parsed by later
+ */
+const getSchedule = config => {
+  if (!config) {
+    return;
+  }
+  if (config.text_expression) {
+    // text expression takes precedence over cron
+    return later.parse.text(config.text_expression);
+  }
+  if (config.cron) {
+    return later.parse.cron(config.cron);
+  }
+};
+
+/**
+ * Return the milliseconds ahead for the first job scheduled.
+ * If the first job time is less than 1 second ahead of time,
+ * add a thousand milliseconds to the response.
+ * @param scheduleConfig a scheduling configuration parsed by "later"
+ * @return {number} milliseconds ahead
+ */
+const nextScheduleMillis = scheduleConfig => {
+  const schedule = later.schedule(scheduleConfig);
+  const diff = schedule.next().getTime() - Date.now();
+  if (diff <= 1000) {
+    return diff + 1000;
```

Comment:
:thinking: yes actually is not necessary, because even for very short difference of milliseconds now that we use the native `setTimeout()` method, the schedule will always run, regardless if the Node virtual machine can actually execute it on time or because high workload it does milliseconds / seconds after. I'll remove the the addition and adapt the tests.